java/flight/flight-sql-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/ArrowFlightPreparedStatement.java

```diff
@@ -93,6 +104,17 @@ public synchronized void close() throws SQLException {
 
   @Override
   public FlightInfo executeFlightInfoQuery() throws SQLException {
+    List<FieldVector> fields = new ArrayList<>();
+    for(int i = 0; i < this.getParameterCount(); i++) {
+      AvaticaParameter param = this.getParameter(i);
+      FieldVector v = switch (param.parameterType) {
+        case DECIMAL:
+          new DecimalVector()
+      };
+      FieldVector v = new
+    }
+    VectorSchemaRoot parameters = new VectorSchemaRoot();
+    this.preparedStatement.setParameters();
```

Comment:
I think this is the missing piece. If we can translate the `AvaticaParameter`s into a `VectorSchemaRoot` the server will get what it needs.